pkg/secrets/pushtofile/secret_group.go

```diff
@@ -85,12 +86,29 @@ func (s *SecretGroup) pushToFileWithDeps(
 		_ = wc.Close()
 	}()
 
-	return depPushToWriter(
+	// Calls to pushToWriter include a second-pass at parsing and executing the provided Go template
+	// First-pass parsing and rendering is validated with dummy secret values in newSecretGroup
+	// Output from this step should be redirected to /dev/null to mitigate secret leaks in log output
+	// This may result in obscured details of an execution failure here, which would require further experimentation in a local environment
+	stdout := os.Stdout
+	stderr := os.Stderr
+	defer func() {
+		os.Stdout = stdout
+		os.Stderr = stderr
+	}()
+	os.Stdout = os.NewFile(0, os.DevNull)
+	os.Stderr = os.NewFile(0, os.DevNull)
```

Comment:
Some thoughts...

Redirecting to `/dev/null` in-process will be in effect while this function runs to completion. Code executing in other Go routines at the same time will be affected. I don't know if there's way to limit the effect to the function body.

---

I think there are other approaches to preventing secret leakage that don't need to rely on redirecting to `/dev/null`...

Here are the ways in which secrets could be leaked 
1.`depPushToWriter` panics, and prints to stdout/stderr
2. `depPushToWriter` returns an error, which is then printed to stdout/stderr
3. `depPushToWriter` uses the log module and leaks to stdout/stderr

(1) and (2) and be dealt with in a similar fashion. (1) requires recovering from the panic first. The approach is to hide the actual error and instead return a generic error that states that write to file failed due to issues executing the template.

For (3) we can statically find all the places in the code where `log.RecordedErr` etc. are invoked, and we can ensure the payload doesn't include a potential leak.

Reply:
1. Added panic recovery before call to `depPushToWriter`
2. Errors returned by `depPushToWriter` are received, and then exchanged for a hardcoded message
3. The standard `depPushToWriter`, `pushToWriter`, does not use `log.RecordedError`